src/mpid/ch4/netmod/ofi/ofi_rma.h

```diff
@@ -453,6 +456,25 @@ static inline int MPIDI_NM_mpi_put(const void *origin_addr,
                                               + target_true_lb, MPIDI_OFI_winfo_mr_key(win,
                                                                                        target_rank)),
                               rdma_inject_write);
+    } else if (origin_contig && target_contig) {
+        offset = target_disp * MPIDI_OFI_winfo_disp_unit(win, target_rank);
+        MPIR_ERR_CHKANDJUMP((origin_bytes != target_bytes), mpi_errno, MPI_ERR_SIZE, "**rmasize");
+
+        msg.desc = NULL;
+        msg.addr = MPIDI_OFI_comm_to_phys(win->comm_ptr, target_rank);
+        msg.context = NULL;
+        msg.data = 0;
+        msg.msg_iov = &iov;
+        msg.iov_count = 1;
+        msg.rma_iov = &riov;
+        msg.rma_iov_count = 1;
+        iov.iov_base = (char *) origin_addr + origin_true_lb;
+        iov.iov_len = target_bytes;
+        riov.addr = (uint64_t) (MPIDI_OFI_winfo_base(win, target_rank) + offset + target_true_lb);
+        riov.len = target_bytes;
+        riov.key = MPIDI_OFI_winfo_mr_key(win, target_rank);
+        MPIDI_OFI_CALL_RETRY2(MPIDI_OFI_win_cntr_incr(win),
+                              fi_writemsg(MPIDI_OFI_WIN(win).ep, &msg, 0), rdma_write);
     } else {
         mpi_errno = MPIDI_OFI_do_put(origin_addr,
                                      origin_count,
```

Comment:
`MPIDI_NM_mpi_rput` uses `MPIDI_OFI_do_put ` to transfer all remote PUT operations. Thus, the same performance issue should exist. Should we implement the continuous put optimization in `MPIDI_OFI_do_put` ?

Similar issue in `MPIDI_NM_mpi_rget`.

Reply:
Yes, the issue exists for both `MPIDI_NM_mpi_rput` and `MPIDI_NM_mpi_rget`. Do you think we should further modify those two, or should optimize `MPIDI_OFI_do_put` and `MPIDI_OFI_do_get` instead ?

Reply:
i agree